synapse/handlers/account_validity.py

```diff
@@ -299,3 +299,22 @@ def _mark_expired_users_as_inactive(self):
 
         # Mark each as non-active
         yield self.profile_handler.set_active(active_expired_users, False, True)
+
+
+@defer.inlineCallbacks
+def is_account_expired(user_id: str, store, current_ts: int):
```

Comment:
We generally try not to pass the `store` around between functions, but instead always reference it from the homeserver object.

However, I think it would make more sense if this method were moved next to `get_expiration_ts_for_user` in `synapse/storage/data_stores/main/registration.py`. Then you could simply do `yield store.is_account_expired(...)`, rather than having to import `is_account_expired` everywhere you want to use it.